src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,277 @@
     public T remove(int index) {
-        return null;
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("Index " + index + " does not exist");
+        }
+
+        Node<T> elementToRemove = null;
+
+        if (size == 1) {
+            elementToRemove = head;
+            head = null;
+            tail = null;
+        } else if (index == size - 1) {
+            elementToRemove = tail;
+            Node<T> elPrevElementToRemove = tail.prev;
+            elPrevElementToRemove.next = null;
+            tail = elPrevElementToRemove;
+        } else if (index == 0) {
+            elementToRemove = head;
+            Node<T> elNextElementToRemove = head.next;
+            elNextElementToRemove.prev = null;
+            head = elNextElementToRemove;
+        } else {
+            int countOfIndex = 0;
+            Node<T> element = head;
+
+            while (countOfIndex != index) {
+                element = element.next;
+                countOfIndex++;
+            }
+
+            elementToRemove = element;
+            Node<T> elPrevElementToRemove = elementToRemove.prev;
+            Node<T> elNextElementToRemove = elementToRemove.next;
+
+            elPrevElementToRemove.next = null;
+            elNextElementToRemove.prev = null;
+
+            elPrevElementToRemove.next = elNextElementToRemove;
+            elNextElementToRemove.prev = elPrevElementToRemove;
+        }
+
+        size--;
+        return elementToRemove.item;
     }
 
     @Override
     public boolean remove(T object) {
+        boolean isExist = false;
+
+        if (size == 0) {
+            isExist = false;
+        } else if (size == 1 && object.equals(head.item)) {
+            head = null;
+            tail = null;
+            isExist = true;
+        } else if (size > 1) {
+            int countOfIndex = 0;
+            Node<T> element = head;
+
+            while (countOfIndex != size) {
+                if (element.item == null ? object == null : element.item.equals(object)) {
+                    isExist = true;
+                    break;
+                }
+                element = element.next;
+                countOfIndex++;
+            }
+
+            if (isExist) {
+                if (countOfIndex == size - 1) {
+                    Node<T> elPrevElement = element.prev;
+                    elPrevElement.next = null;
+                    tail = elPrevElement;
+                } else if (countOfIndex == 0) {
+                    Node<T> elNextElement = element.next;
+                    elNextElement.prev = null;
+                    head = elNextElement;
+                } else {
+                    Node<T> elPrevElement = element.prev;
+                    Node<T> elNextElement = element.next;
+
+                    elPrevElement.next = null;
+                    elPrevElement.next = elNextElement;
+
+                    elNextElement.prev = null;
+                    elNextElement.prev = elPrevElement;
```

Comment:
In the `remove(T object)` method, setting `next` and `prev` to `null` before updating the pointers to bypass the node is redundant and incorrect. This can lead to breaking the list structure.